paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -294,33 +293,40 @@ internal class CustomerSheetViewModel @Inject constructor(
         formViewData: FormViewModel.ViewData,
     ) {
         viewModelScope.launch {
-            runCatching {
-                val params = formViewData.completeFormValues
-                    ?.transformToPaymentMethodCreateParams(paymentMethodSpec)
-                val paymentMethod = requireNotNull(createPaymentMethod(params))
-                attachPaymentMethodToCustomer(paymentMethod)
-            }.onFailure {
-                updateViewState<CustomerSheetViewState.AddPaymentMethod> {
-                    // TODO (jameswoo) translate message
-                    it.copy(
-                        errorMessage = "Could not create payment method",
-                        isProcessing = false,
+            if (formViewData.completeFormValues == null) error("completeFormValues cannot be null")
+            val params = formViewData.completeFormValues
+                .transformToPaymentMethodCreateParams(paymentMethodSpec)
+            createPaymentMethod(params)
+                .onSuccess { paymentMethod ->
+                    attachPaymentMethodToCustomer(paymentMethod)
+                }.onFailure { throwable ->
+                    logger.error(
+                        msg = "Failed to create payment method for $paymentMethodSpec",
+                        t = throwable,
                     )
+                    updateViewState<CustomerSheetViewState.AddPaymentMethod> {
+                        it.copy(
+                            errorMessage = (throwable as? StripeException)?.stripeError?.message
```

Comment:
Another one of these.

Also, should we pass context into the helper and do this something when wrong by default?